aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -33,16 +39,19 @@ class OperatorEntry final {
 
   void prepareForDeregistration();
 
-  RegistrationHandleRAII registerKernel(c10::optional<DispatchKey> dispatch_key, KernelFunction kernel);
+  // Postcondition: caller is responsible for disposing of the kernel
+  std::list<KernelFunction>::iterator registerKernel(c10::optional<DispatchKey> dispatch_key, KernelFunction kernel);
+  void deregisterKernel_(c10::optional<DispatchKey> dispatch_key, std::list<KernelFunction>::iterator kernel);
 
   void updateSchemaAliasAnalysis(AliasAnalysisKind a) {
-    schema_.setAliasAnalysis(a);
+    TORCH_INTERNAL_ASSERT(schema_.has_value());
+    schema_->setAliasAnalysis(a);
   }
 
 private:
-  void deregisterKernel_(c10::optional<DispatchKey> dispatch_key, std::list<KernelFunction>::iterator kernel);
 
-  FunctionSchema schema_;
+  OperatorName name_;
+  c10::optional<FunctionSchema> schema_;
```

Comment:
Does it have to be optional? Can we just fill in an inferred schema from the kernel function if we only have impl calls?

Reply:
No, because you might have registered a boxed function and in that case there's no impl. Also it doesn't match the spec I wrote ;)